packages/cws-trx/src/sign.ts

```diff
@@ -22,49 +25,66 @@ export const signTransaction = async (
   signTxData: any,
   script: string,
   argument: string,
-  publicKey: string | undefined = undefined,
+  publicKey: string
 ): Promise<string> => {
-
-  const { transport, transaction } = signTxData
-
-  const rawPayload = trxUtil.getRawHex(transaction);
+  const {
+    transport, appPrivateKey, appId, confirmCB, authorizedCB
+  } = signTxData;
 
   const preActions = [];
-  let action;
   const sendScript = async () => {
     await apdu.tx.sendScript(transport, script);
-  }
+  };
+
   preActions.push(sendScript);
 
-  action = async () => {
-    return apdu.tx.executeScript(
-      transport,
-      signTxData.appId,
-      signTxData.appPrivateKey,
-      trxUtil.getArgument(signTxData)
-    );
-  }
+  const action = async () => apdu.tx.executeScript(
+    transport,
+    appId,
+    appPrivateKey,
+    argument
+  );
+
   const canonicalSignature = await tx.flow.getSingleSignatureFromCoolWallet(
     transport,
     preActions,
     action,
     false,
-    signTxData.confirmCB,
-    signTxData.authorizedCB,
+    confirmCB,
+    authorizedCB,
     true
   );
 
-  const { signedTx } = await apdu.tx.getSignedHex(transport);
+  if (Buffer.isBuffer(canonicalSignature)) return '';
+	const {r, s} = canonicalSignature;
 
-  if (!Buffer.isBuffer(canonicalSignature)) {
-    const { v, r, s } = await trxUtil.genTrxSigFromSESig(
-      canonicalSignature,
-      rawPayload,
-      publicKey
-    );
-    const serializedTx = trxUtil.composeSignedTransacton(signTxData, v, r, s);
-    return serializedTx;
-  } else {
-    throw new error.SDKError(signTransaction.name, 'canonicalSignature type error');
-  }
+  const {signedTx} = await apdu.tx.getSignedHex(transport);
+	console.log('signedTx :', signedTx);
```

Comment:
I'd suggest to remove most `console.log` unless we have a good reason. Or, we shall assign various log levels. e.g., error, warning, info, debug, etc.